### PR TITLE
chore: enable `strictFunctionTypes` in tsconfig.json

### PR DESCRIPTION
**Pull request by kswenson**
"Fixes" fall into three broad categories:
- actual fixes to the typing issues in question
- adding `@ts-expect-error` at the point of the call in places where the actual fix wasn't clear
- ~~adding a cast in some places where `@ts-expect-error` didn't work or was annoying~~

Update: I figured out the root of the issues we were having with D3 types. The trick is to type the initial selection correctly because that determines the types of all of the arguments from the chained callbacks. Also, the functions passed to `join()` must return the appropriate type to enable further chaining.

### CODE REVIEW

File: v3/src/components/graph/components/background.tsx [hunk 1]
```diff
@@ -123,6 +124,7 @@ export const Background = forwardRef<SVGGElement, IProps>((props, ref) => {
         (enter) => {
           enter.append('rect')
             .attr('class', 'graph-background')
+            // @ts-expect-error strictFunctionTypes
```
Comment (emcelroy) [inline v3/src/components/graph/components/background.tsx:127]: FWIW: in CLUE, this wasn't needed after I changed line 109 to `dragBehavior = drag<SVGRectElement, number>()`.

**Review by emcelroy — APPROVED** [review]
Looks good 👍

**Review by bfinzer — APPROVED** [review]
👍🏻LGTM
Amazing!

**Review by emcelroy — APPROVED** [review]
Looks even better!

**Pull request merged**